### ISSUE TITLE
errors on page transition

### ISSUE THREAD

**Issue by jzimmek** [nareshbhatia/mobx-state-router#28]
hi, i really like the approach of mobx-state-router and tested it out for our upcoming, mobx based, project. it was a good experience so far, but i run into two issues:

the following error is thrown the first time i transition to a different url:

```
19:02:28.405 router-store.js:102 RouterStore.transition({"routeName":"home","params":{},"queryParams":{}}, {"routeName":"choose","params":{},"queryParams":{}})
19:02:28.414 history-adapter.js:17 HistoryAdapter.goToLocation({"pathname":"/choose","search":"","hash":"","key":"e27ycl"})
19:02:28.423 mobx.module.js:3516 [mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Reaction@20] TypeError: b.valueOf is not a function
    at valueEqual (index.js:21)
    at index.js:31
    at Array.every (<anonymous>)
    at Object.valueEqual [as default] (index.js:30)
    at RouterState.push.8VVh.RouterState.isEqual (router-store.js:31)
    at RouterStore.push.8VVh.RouterStore.transition (router-store.js:90)
    at RouterStore.push.8VVh.RouterStore.goTo (router-store.js:61)
    at HistoryAdapter.goToLocation (history-adapter.js:32)
    at history-adapter.js:60
    at listener (createTransitionManager.js:54)

```

it seems to have no obvious negative effect, because the router shows the expected route.

but i also get the following error when invoking routerStore.goTo:

```
Unhandled promise rejection Error: toState is undefined
    at router-store.js:122
    at run (es6.promise.js:75)
    at es6.promise.js:92
    at MutationObserver.flush (_microtask.js:18)
```

the page transition does not happen. this error disappear if i force reload the current route in the browser and invoke routerStore.goTo.

my environment:

mobx-state-router: 3.0.3
mobx: 4.3.0

i am not sure if those 2 errors are related or not. just let me know and i will create extra issues.

any help would be really appreciated.

**Comment by kutuluk** [comment]
I know the problem and tomorrow I will solve it. While using version 3.0.2

**Comment by jzimmek** [comment]
just tried out 3.0.2 and it works without having any of the above issues. thx.

**Comment by nareshbhatia** [comment]
@kutuluk, by any chance did you include this fix as part of PR #27? If yes, I would appreciate a separate PR for this issue. I prefer commits to address one issue or feature at a time.

Thanks.

**Comment by kutuluk** [comment]
Ok, done =)
